tests/group_by_test.py

```diff
@@ -584,3 +584,10 @@ def test_group_by_selector(constructor: Constructor) -> None:
     )
     expected = {"a": [1, 1], "b": [4, 6], "c": [8.0, 9.0]}
     assert_equal_data(result, expected)
+
+
+def test_renaming_edge_case(constructor: Constructor) -> None:
+    data = {"a": [0, 0, 0], "_a_tmp": [1, 2, 3], "b": [4, 5, 6]}
+    result = nw.from_native(constructor(data)).group_by(nw.col("a")).agg(nw.all().min())
+    expected = {"a": [0], "_a_tmp": [1], "b": [4]}
+    assert_equal_data(result, expected)
```

Comment:
It should be safe as it will have at least one additional character i.e. the new column generated is `_a_tmp_`

Reply:
😄 I wanted a test so that if anyone tried removing the `+1` then this would stop them

Reply:
Yep 100% worth it to have edge case tests